internal/command/init.go

```diff
@@ -808,6 +815,571 @@ func (c *InitCommand) getProviders(ctx context.Context, config *configs.Config,
 	return true, false, diags
 }
 
+// getProvidersFromConfig determines what providers are required by the given configuration data.
+// The method downloads any missing providers that aren't already downloaded and then returns
+// dependency lock data based on the configuration.
+// The dependency lock file itself isn't updated here.
+func (c *InitCommand) getProvidersFromConfig(ctx context.Context, config *configs.Config, upgrade bool, pluginDirs []string, flagLockfile string, view views.Init) (output bool, resultingLocks *depsfile.Locks, diags tfdiags.Diagnostics) {
+	ctx, span := tracer.Start(ctx, "install providers")
```

Comment:
Nitpick: I think we should use a unique span name here, i.e. different from the one already in use above by `getProviders` method and different from `getProvidersFromState` below